### PR TITLE
Added e2e tests for CAL source

### PR DESCRIPTION
**Pull request by capri-xiyue**
## Proposed Changes

- Added e2e tests for CAL source

### CODE REVIEW

File: test/e2e/test_cloudauditlog.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		Version:  "v1",
+		Resource: "jobs",
+	}
+
+	if err := client.WaitForResourceReady(client.Namespace, calName, gvr); err != nil {
+		t.Error(err)
+	}
+
+	//CAL source misses the topic which gets created shortly after CAL source becomes ready. Need to wait for a few seconds
+	time.Sleep(45 * time.Second)
```
Comment (capri-xiyue) [inline test/e2e/test_cloudauditlog.go:101]: @ian-mi  When I did the e2e test, I found CAL source misses the topic which gets created shortly after CAL source becomes ready. Not sure whether it's a big issue(maybe related to the SLO of logsink). For example, when the log sink got created and CAL source becomes ready at 1:42:35pm and one AuditLog got fired at 1:43:04pm, the AuditLog will get missed by the CAL source.

**Review by capri-xiyue — COMMENTED** [review]
When I did the e2e test, I found CAL source misses the CloudAuditLogs which gets created shortly after CAL source becomes ready. Not sure whether it's a big issue(maybe related to the SLO of logsink). For example, when the log sink got created and CAL source becomes ready at 1:42:35pm and one AuditLog got fired at 1:43:04pm, the AuditLog will get missed by the CAL source.

File: test/e2e/test_cloudauditlog.go (hunk 1 above)
Comment (ian-mi) [reply test/e2e/test_cloudauditlog.go:101]: Interesting, I expect that this is caused by eventual consistency of the Stackdriver sink configuration but we should investigate to make sure we are not marking the source as ready too early.

File: test/e2e/config/cloudauditlog_test/target.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+spec:
+  selector:
+    targetUID: {{ .targetUID }}
+  ports:
+    - name: http
+      port: 80
+      protocol: TCP
+      targetPort: 8080
+  sessionAffinity: None
+  type: ClusterIP
```
Comment (ian-mi) [inline test/e2e/config/cloudauditlog_test/target.yaml]: nit: please add newline at end of file

File: test/e2e/config/cloudauditlog_test/cloudauditlog.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  namespace: {{ .namespace }}
+spec:
+  serviceName: {{ .serviceName }}
+  methodName:  {{ .methodName }}
+  resourceName:  {{ .resourceName }}
+  sink:
+    ref:
+      apiVersion: v1
+      kind: Service
+      name: {{ .targetName }}
```
Comment (ian-mi) [inline test/e2e/config/cloudauditlog_test/cloudauditlog.yaml]: nit: please add newline at end of file

**Review by ian-mi — APPROVED** [review]

File: test/e2e/config/cloudauditlog_test/cloudauditlog.yaml (hunk 3 above)
Comment (capri-xiyue) [reply test/e2e/config/cloudauditlog_test/cloudauditlog.yaml]: Done

File: test/e2e/config/cloudauditlog_test/target.yaml (hunk 2 above)
Comment (capri-xiyue) [reply test/e2e/config/cloudauditlog_test/target.yaml]: Done

**Pull request merged**